TYPING_GUIDELINES.md

```diff
@@ -0,0 +1,63 @@
+# Guidelines When adding types 
+
+If a component has subcomponents, e.g `Step` component, which has:
+- `StepContainer`
+- `StepNext`
+- `StepPrevious`
+- ...
+
+It is prefered to use Object.assign to attach them to the main component in the index.ts file:
+
+```tsx
+import Step from "./Step";
+
+export * from "./Step";
+export * from "./StepContainer";
+export * from "./StepNext";
+export * from "./StepPrevious";
+
+export default Object.assign(Step, {
+  Container: StepContainer,
+  Next: StepNext,
+  Previous: StepPrevious
+});
+```
+---
+
+Prop types should be exported `interfaces` named as `${ComponentName}Props` 
+```ts
+export interface DividerProps {
+  vertical?: boolean;
+}
+```
+---
+
+Always export `default` and `*` from component files (that will export types):
+
+```ts
+import Divider from "./Divider";
+export * from "./Divider";
+export default Divider;
+```
+---
+
+If it's a component that renders another component (wrapper), we must extend props interface:
+```tsx
+import Box, { BoxProps } from "../Box";
+export interface DividerProps extends BoxProps {
+  vertical?: boolean;
+}
+
+const DividerComponent = (props: DividerProps) => <Box {...props} />;
+```
+---
+
+When you have: 
+```js
+const component = (props: CodeProps) => {...}
+```
```

Comment:
Let's remove the example that doesn't exist anymore and describe the problem. 

Maybe something like:

If a styled component needs another component declared in the same file (for lifecycles, computed props etc.), it should be named `${ComponentName}Component`:

```js
const CodeComponent = (props: CodeProps) => {...}

const Code = styled(CodeComponent)``
```

Reply:
`CodeComponent` is a bad example because that's the only one that isn't passed to `Code` directly. Let me find a better one.

Reply:
Ok, the classic Hidden:

```ts
const HiddenComponent = (props: HiddenProps) => {...}

const Hidden = styled(HiddenComponent)``
```

Reply:
done!